persistence-cassandra-4.0/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -102,4 +157,184 @@ public ResultMessage processBatch(
     return QueryProcessor.instance.processBatch(
         batchStatement, queryState, options, customPayload, queryStartNanoTime);
   }
+
+  private void authorizeByToken(ByteBuffer token, CQLStatement statement) {
+    String authToken = StandardCharsets.UTF_8.decode(token).toString();
+
+    if (!getAuthorizationService().isPresent()) {
+      throw new RuntimeException(
+          "Failed to find an io.stargate.auth.AuthorizationService to authorize request");
+    }
+
+    AuthorizationService authorization = getAuthorizationService().get();
+    if (statement instanceof SelectStatement) {
```

Comment:
Is this common enough across the 3 persistence implementations to be worth a common / shared method or class?

Reply:
That's about the only thing common between them, 1) decode token, 2) check if authService is present, 3) get the authService. Is four lines of code worth a new class?

Reply:
Ah, ok, sorry, I did not compare the code carefully :facepalm: